OrbitCore/FunctionUtils.cpp

```diff
@@ -0,0 +1,197 @@
+      "FAR_FAST",     // CV_CALL_FAR_FAST    = 0x05, // far left to right push
+                      // with regs, callee pops stack
+      "SKIPPED",   // CV_CALL_SKIPPED     = 0x06, // skipped (unused) call index
+      "NEAR_STD",  // CV_CALL_NEAR_STD    = 0x07, // near standard call
+      "FAR_STD",   // CV_CALL_FAR_STD     = 0x08, // far standard call
+      "NEAR_SYS",  // CV_CALL_NEAR_SYS    = 0x09, // near sys call
+      "FAR_SYS",   // CV_CALL_FAR_SYS     = 0x0a, // far sys call
+      "THISCALL",  // CV_CALL_THISCALL    = 0x0b, // this call (this passed in
+                   // register)
+      "MIPSCALL",  // CV_CALL_MIPSCALL    = 0x0c, // Mips call
+      "GENERIC",   // CV_CALL_GENERIC     = 0x0d, // Generic call sequence
+      "ALPHACALL",  // CV_CALL_ALPHACALL   = 0x0e, // Alpha call
+      "PPCCALL",    // CV_CALL_PPCCALL     = 0x0f, // PPC call
+      "SHCALL",     // CV_CALL_SHCALL      = 0x10, // Hitachi SuperH call
+      "ARMCALL",    // CV_CALL_ARMCALL     = 0x11, // ARM call
+      "AM33CALL",   // CV_CALL_AM33CALL    = 0x12, // AM33 call
+      "TRICALL",    // CV_CALL_TRICALL     = 0x13, // TriCore Call
+      "SH5CALL",    // CV_CALL_SH5CALL     = 0x14, // Hitachi SuperH-5 call
+      "M32RCALL",   // CV_CALL_M32RCALL    = 0x15, // M32R Call
+      "CLRCALL",    // CV_CALL_CLRCALL     = 0x16, // clr call
+      "INLINE",     // CV_CALL_INLINE      = 0x17, // Marker for routines always
+                    // inlined and thus lacking a convention
+      "NEAR_VECTOR",  // CV_CALL_NEAR_VECTOR = 0x18, // near left to right push
+                      // with regs, callee pops stack
+      "RESERVED"};    // CV_CALL_RESERVED    = 0x19  // first unused call
+                      // enumeration
+  if (func.CallingConvention() < 0 ||
+      static_cast<size_t>(func.CallingConvention()) >=
+          (sizeof(kCallingConventions) / sizeof(kCallingConventions[0]))) {
+    return "UnknownCallConv";
+  }
+
+  return kCallingConventions[func.CallingConvention()];
+}
+
+uint64_t Offset(const Function& func) {
+  return func.Address() - func.LoadBias();
+}
+
+bool IsOrbitFunc(const Function& func) {
+  return func.GetOrbitType() != Function::OrbitType::NONE;
+}
+
+bool Hookable(const Function& func) {
+  if (Capture::IsLinuxData()) {
+    return true;
+  } else {
+    // Don't allow hooking in asm implemented functions (strcpy, stccat...)
+    // TODO: give this better thought.  Here is the theory:
+    // Functions that loop back to first 5 bytes of instructions will explode as
+    // the IP lands in the middle of the relative jump instruction...
+    // Ideally, we would detect such a loop back and not allow hooking.
+    if (func.File().find(".asm") != std::string::npos) {
+      return false;
+    }
+
+    CV_call_e conv = static_cast<CV_call_e>(func.CallingConvention());
+    return ((conv == CV_CALL_NEAR_C || conv == CV_CALL_THISCALL) &&
+            func.Size() >= 5);
+  }
+}
+
+void Select(Function& func) {
```

Comment:
here and below. Function* func (use pointers for output parameters)

Reply:
Done, thanks!